models/articles.models.js

```diff
@@ -14,32 +14,49 @@ exports.selectArticleById = (article_id) => {
     });
 };
 
-exports.selectArticles = (sort_by, order) => {
+exports.selectArticles = (sort_by, order, topic) => {
+  const queryProms = [];
+  const queryVals = [];
   let queryStr =
-    "SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id GROUP BY articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url";
+    "SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id";
 
-  const validColumns = ["author", "title", "article_id", "topic", "created_at", "votes", "article_img_url"]
+  if (topic) {
+    queryProms.push(checkValueExists("articles", "topic", topic));
+    queryVals.push(topic);
+    queryStr +=
+      " WHERE topic = $1 GROUP BY articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url";
+  } else {
+    queryStr +=
+      " GROUP BY articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url";
+  }
+  if (sort_by) {
+    queryProms.push(checkColumnExists("articles", sort_by));
 
-  if(sort_by){
-    if(!validColumns.includes(sort_by, order)){
-      return Promise.reject({ status: 400, message : 'Bad request'})
-    } else {
-      queryStr += ` ORDER BY ${sort_by};`
-    }
+    queryStr += ` ORDER BY ${sort_by};`;
   } else {
-    queryStr += ` ORDER BY created_at;`
+    queryStr += ` ORDER BY created_at;`;
   }
 
-  if(order){
-    order = order.toUpperCase()
-    queryStr=queryStr.slice(0, -1)
-    queryStr+= ` ${order};`
+  if (order) {
+    if (!order === "asc" && !order === "desc") {
+      return Promise.reject({ status: 400, message: "Bad request" });
+    } else {
+      order = order.toUpperCase();
+      queryStr = queryStr.slice(0, -1);
+      queryStr += ` ${order};`;
+    }
```

Comment:
If you are checking the order is exactly asc or desc above then I don't know why you'd need to slice it